pkg/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,69 @@
+package bootstrap
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/weaveworks/pctl/pkg/runner"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	"gopkg.in/yaml.v2"
+)
+
+type Config struct {
+	// GitRepository is the git repository flux resource the installation uses
+	GitRepository profilesv1.GitRepository `json:"gitRepository,omitempty"`
+}
+
+var r runner.Runner = &runner.CLIRunner{}
+
+func CreateConfig(namespace, name, directory string) error {
+	out, err := r.Run("git", "-C", directory, "rev-parse", "--show-toplevel")
+	if err != nil {
+		if strings.Contains(string(out), "not a git repository") {
+			return fmt.Errorf("the target directory %q is not a git repository", directory)
+		}
+		return fmt.Errorf("failed to get git directory location: %w", err)
+	}
+
+	pctlDir := filepath.Join(strings.TrimSuffix(string(out), "\n"), ".pctl")
+	if err := os.Mkdir(pctlDir, 0755); err != nil {
+		return fmt.Errorf("failed to create .pctl dir %q: %w", pctlDir, err)
+	}
+
+	data, err := yaml.Marshal(Config{
+		GitRepository: profilesv1.GitRepository{
+			Name:      name,
+			Namespace: namespace,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to marshal config: %w", err)
+	}
+
+	return os.WriteFile(filepath.Join(pctlDir, "config.yaml"), data, 0644)
+}
+
+func GetConfig(directory string) (*Config, error) {
```

Comment:
And here ofc. :D

Reply:
We aren't modifying Config and you aren't checking nil on it either, so there is no need to share Config up the stack. Remove the pointer, it doesn't need to be one.

Reply:
its a pointer because the config might not exist, and no error occurs, so we need a way of returning "I didn't error, but the config doesn't exist", so having it be nil is the simplest solution.

Reply:
Sure, but you aren't checking if it's nil then.

```go
		config, err := bootstrap.GetConfig(wd)
		if err == nil {
			gitRepoNamespace = config.GitRepository.Namespace
			gitRepoName = config.GitRepository.Name
		}
```

Which means it will error out.

Also, every path you have in GetConfig returns with an error if the file can't be accessed.

Reply:
And now that I think about it, the error is disregarded, right? We don't log it in any format anywhere. Or am I missing something here / just can't read? :)

Reply:
oops this is missing haha. should be `if err == nil && config != nil`. Because configs are optional, I wasn't really sure how to handle the error. Ideally we'd have a debug log level to print too, but we don't 😕